hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -476,17 +475,21 @@ public Response initializeMultipartUpload(
       OzoneBucket ozoneBucket = getBucket(bucket);
       String storageType = headers.getHeaderString(STORAGE_CLASS_HEADER);
 
-      S3StorageType s3StorageType;
-      if (storageType == null || storageType.equals("")) {
-        s3StorageType = S3StorageType.getDefault(ozoneConfiguration);
-      } else {
-        s3StorageType = S3Utils.toS3StorageType(storageType);
+      ReplicationConfig clientConfiguredReplicationConfig = null;
+      String replication = ozoneConfiguration.get(OZONE_REPLICATION);
+      if (replication != null) {
+        clientConfiguredReplicationConfig = ReplicationConfig.parse(
+            ReplicationType.valueOf(ozoneConfiguration
+                .get(OZONE_REPLICATION_TYPE, OZONE_REPLICATION_TYPE_DEFAULT)),
+            replication, ozoneConfiguration);
       }
-      ReplicationType replicationType = s3StorageType.getType();
-      ReplicationFactor replicationFactor = s3StorageType.getFactor();
+      ReplicationConfig replicationConfig = S3Utils
+          .resolveS3ClientSideReplicationConfig(storageType,
+              clientConfiguredReplicationConfig,
+              ozoneBucket.getReplicationConfig());
```

Comment:
Nit: can this block be extracted to a function instead of duplicating it from `put`?

Reply:
Thanks @adoroszlai for the review. I have just updated the patch by fixing it.